plugins/Toolbox/resources/qml/ToolboxInstalledTile.qml

```diff
@@ -195,10 +188,13 @@ Item
         ProgressBar
         {
             id: progressbar
-            anchors.left: updateButton.left
-            anchors.right: updateButton.right
-            anchors.top: updateButton.bottom
-            anchors.topMargin: 4
+            anchors
+            {
+                left: updateButton.left
+                right: updateButton.right
+                top: updateButton.bottom
+                topMargin: Math.floor(UM.Theme.getSize("base_unit") / 4)
```

Comment:
Use the default margin?

Reply:
Too big. Needs to be 0.25 units, default margin is 1. Could define yet another margin on a case by case basis but don't really see the point of polluting the theme with that.

Reply:
>  don't really see the point of polluting the theme with that.

But that's the whole point of theming. By hardcoding the 4, you basically make it unthemeable.

I would argue that if just one element needs this adjusted margin, then the design is using margins inconsistently.

Reply:
>I would argue that if just one element needs this adjusted margin, then the design is using margins inconsistently.

Luke is quite good about keeping things consistent. This is one spot in the code but many places in the interface.

What I mean about polluting the theme is what I see as making it too granular. Since the theme doesn't cascade and can't inherit element-specific measurements from more general ones, we end up having to make a trade off between "what if I want to generally change one size, then I have to try to figure out everywhere where that size is used and update 10 different values in the theme because each one is defined separately" which is annoying to some vs. "what if I want to make everything a random size untethered from everything else, but I can't because everything is proportional to each other" which would be annoying to others.

I'd love to theme Cura but I won't be investing the time into it because I find it incredibly bothersome because it's so granular. I can change these to all have their own definition but I don't think it's definitely not UM.Theme "more useful."

My plan for the research sprint is to re-write how lots of UI stuff is handled so that it can have cascading/inheritance, comments, x-ref'd values, etc.